aiidalab_launch/instance.py

```diff
@@ -128,7 +141,10 @@ def configuration_changes(self) -> Generator[str, None, None]:
 
         try:
             for mount in self._mounts():
-                if docker_mount_for(self.container, mount["Target"]) != mount["Source"]:
+                if (
```

Comment:
This is a bugfix unrelated to the changes in this PR. `docker_mount_for` returns Path for bind mounts, while `mount["Source"]` is always str so this check would never be true even though the configuration did not change, and the user would get spurious warnings. LMK if you want me submit a separate PR for this.

Reply:
I would be ideal to fix this bug in a separate PR. There we could also try to add a simple test that catches this bug. This function is tested [pretty extensively](https://github.com/aiidalab/aiidalab-launch/blob/73fe854e525d1c0adfa1f92b1aa97842df5a5c16/tests/test_instance.py#L78) – so I would think that the undesired behavior should show up there.

Reply:
@csadorf submitted a new PR #134 and added a test. I will revert this change on this PR and rebase/merge with main after #134 is merged.